agimus_controller/agimus_controller/factory/robot_model.py

```diff
@@ -135,15 +162,36 @@ def load_models(self) -> None:
     def _load_full_pinocchio_models(self) -> None:
         """Load the full robot model, the visual model and the collision model."""
         try:
-            (
-                self._full_robot_model,
-                self._collision_model,
-                self._visual_model,
-            ) = pin.buildModelsFromUrdf(
-                str(self._params.urdf_path),
-                str(self._params.urdf_meshes_dir),
-                pin.JointModelFreeFlyer() if self._params.free_flyer else None,
-            )
+            geometry_types = [
+                pin.GeometryType.COLLISION,
+                (pin.GeometryType.VISUAL if self._params.urdf_meshes_dir else None),
+            ]
```

Comment:
How comes meshes are only needed for visual model? isn't by default that collision shapes match visual shapes?

Reply:
You are actually right. I don't know what I was thinking when implementing it. Good catch